### PR TITLE
tidb-functions: extend tidb_decode_key docs

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Extend documentation for `TIDB_DECODE_KEY`

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

Closes: #5561

### CODE REVIEW

**Review by kolbe — APPROVED** [review]
LGTM

File: functions-and-operators/tidb-functions.md [hunk 1]
```diff
@@ -22,7 +22,9 @@ TableStmt ::=
     "TIDB_DECODE_KEY(" STR ")"
 ```
 
-### Example
+### Examples
+
+In the first example we have a table `t1` which has a hidden rowid that is generated by TiDB. This is what happens for a non-clustered primary key.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:27]:
```suggestion
In the following example, the table `t1` has a hidden `rowid` that is generated by TiDB. The `TIDB_DECODE_KEYSQL` is used in the statement. From the result, you can see that the hidden `rowid` is decoded and output, which is a typical result for the non-clustered primary key.
```

File: functions-and-operators/tidb-functions.md [hunk 2]
```diff
@@ -37,6 +39,69 @@ TIDB_DECODE_KEY(START_KEY): {"_tidb_rowid":1958897,"table_id":"59"}
 1 row in set (0.00 sec)
 ```
 
+In the second example we have a table `t2` that has a compound clustered primary key. Here the JSON output contains a `handle` that contains the name and value for both of the columns that are part of the primary key.
```
Comment (TomShawn) [inline functions-and-operators/tidb-functions.md:42]:
```suggestion
In the following example, the table `t2` has a compound clustered primary key. From the JSON output, you can see a `handle` that contains the name and value for both of the columns that are part of the primary key.
```

File: functions-and-operators/tidb-functions.md (hunk 1 above)
Comment (dveeden) [reply functions-and-operators/tidb-functions.md]: I changed `TIDB_DECODE_KEYSQL` to `TIDB_DECODE_KEY` but otherwise applied the change.

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**